### PR TITLE
DEVPROD-11385 OTel process counters

### PR DESCRIPTION
**Pull request by ybrill**
[DEVPROD-11385](https://jira.mongodb.org/browse/DEVPROD-11385)

### Description
In https://github.com/evergreen-ci/evergreen/pull/8303 it was raised that we don't have visibility into the number of processes that are running. This PR adds process counts in accordance with the spec [here](https://opentelemetry.io/docs/specs/semconv/system/system-metrics/#metric-systemprocesscount).

### Testing
[Works on staging](https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen/result/jWZ9Bkkmztq?omitMissingValues)™️

### CODE REVIEW

**Review by ZackarySantana — APPROVED** [review]
LGTM (I am secondary reviewer, so we need a second review on this from evg-app too)

**Review by Kimchelly — COMMENTED** [review]
The change itself doesn't seem risky, but since we don't have tests for this code and gopsutil is heavily platform-dependent, do we know if the gopsutil will collect these metrics accurately on all platforms? Do we have a plan to check after the deploy that the metrics are being collected as expected on all the platforms of concern? Alternatively, should we add a test to the agent to verify that the call to `ProcessesWithContext` at least returns some kind of reasonable result on all platforms (like at minimum it shows 1 running process)?

**Comment by ybrill** [comment]
@Kimchelly checking is a good idea. They claim to support all the platforms, but we'll see, I guess.
Added a test as well.

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**
